spacy/tokens/doc.pyx

```diff
@@ -285,27 +285,27 @@ cdef class Doc:
             heads = [0] * len(deps)
         if heads and not deps:
             raise ValueError(Errors.E1017)
+        _sent_starts: Optional[List[int]] = [] if sent_starts is not None else None
```

Comment:
Based on the processing below, it looks like `int` values are supported, so I think the original type should be corrected to `Optional[List[Union[bool, int, None]]]` and then I think you can just make a copy and not need to modify the other processing steps:

```python
sent_starts = list(sent_starts) if sent_starts is not None else None
```

Reply:
It would certainly be good to keep `sent_starts` rather than `_sent_starts` lower down in the method, but is performance not a concern at all? That way we make an additional copy of the list.

Reply:
I don't think there's going to be a big difference between creating a copy vs. creating a new list variable. I think it's much simpler to just create a copy to avoid this side effect.